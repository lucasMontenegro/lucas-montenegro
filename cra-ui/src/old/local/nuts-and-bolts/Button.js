import React from 'react';
import styled from 'styled-components';
import palette from './palette';

const createButton = color => styled.button`
  margin: 0 0.5ch;
  border: 0;
  padding: 0.75em 2ch;
  border-radius: 2px;
  color: ${palette[color].contrast}
  background-color: ${palette[color].normal};

  :hover {
    background-image: linear-gradient(transparent,rgba(0,0,0,.05) 40%,rgba(0,0,0,.1));
  }

  :disabled {
    opacity: 0.4;
  }

  :active {
    box-shadow: 0 0 0 1px rgba(0,0,0,.15) inset, 0 0 6px rgba(0,0,0,.2) inset;
  }
`;

const GrayButton = createButton('gray');
const BlueButton = createButton('blue');
const RedButton = createButton('red');
const GreenButton = createButton('green');
const OrangeButton = createButton('orange');

const Button = ({ primary, error, success, warning, children, ...props }) => {
  return primary ? <BlueButton {...props}>{children}</BlueButton>
  : error ? <RedButton {...props}>{children}</RedButton>
  : success ? <GreenButton {...props}>{children}</GreenButton>
  : warning ? <OrangeButton {...props}>{children}</OrangeButton>
  : <GrayButton  {...props}>{children}</GrayButton>
}

export default Button;
