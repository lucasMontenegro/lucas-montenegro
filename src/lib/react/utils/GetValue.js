import React from "react"
import Div from "lib/react/utils/Div"
import Button from "@material-ui/core/Button"
import StringifyObject from "lib/react/utils/StringifyObject"
import PropTypes from "prop-types"
export default function GetValue ({ id, className, value, onClick }) {
  return (
    <Div id={id} className={className} color="black">
      <h5>{id || className}</h5>
      <Div>
        <Button
          className="button"
          variant="outlined"
          onClick={onClick}
        >
          GET VALUE
        </Button>
      </Div>
      <StringifyObject className="value" source={value} />
    </Div>
  )
}
GetValue.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
}