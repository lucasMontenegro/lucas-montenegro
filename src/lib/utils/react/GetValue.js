import React from "react"
import Div from "lib/utils/react/Div"
import Button from "@material-ui/core/Button"
export default function ButtonPlusValue ({ id, className, value, onClick }) {
  return (
    <Div id={id} className={className} color="black">
      <h6>{id || className}</h6>
      <Div>
        <Button
          className="button"
          variant="outlined"
          onClick={onClick}
        >
          GET VALUE
        </Button>
      </Div>
      <Div>Value: <span className="value">{value}</span></Div>
    </Div>
  )
}