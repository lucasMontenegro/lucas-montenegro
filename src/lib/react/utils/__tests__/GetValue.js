import renderer from "react-test-renderer"
import React from "react"
import Button from "@material-ui/core/Button"
import GetValue from "../GetValue"
jest.mock(`@material-ui/core/Button`, () => ({
  __esModule: true,
  default: jest.fn(),
}))
Button.mockImplementation(props => (<button {...props} />))
describe(`../GetValue`, () => {
  it(`should use the right versions of its dependencies`, () => {
    expect(jestUtils.getDependencies([
      `react`,
      `@material-ui/core`,
      `prop-types`,
    ])).toMatchSnapshot()
  })
  {
    const id = `id`
    const className = `className`
    const cases = [
      [{ id, className, }],
      [{     className, }],
      [{ id,            }],
      [{                }],
    ]
    describe.each(cases)(`GetValue (%j)`, props => {
      it(`should render`, () => {
        expect(renderer.create(
          <GetValue
            {...props}
            value={{ bar: 8 }}
            onClick={() => {}}
          />
        ).toJSON()).toMatchSnapshot()
      })
    })
  }
})