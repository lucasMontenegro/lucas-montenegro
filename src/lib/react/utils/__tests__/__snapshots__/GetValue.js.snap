// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`../GetValue GetValue ({"className":"className"}) should render 1`] = `
<div
  className="className"
  style={
    Object {
      "border": "2px solid black",
      "margin": 4,
      "minHeight": undefined,
      "padding": 4,
    }
  }
>
  <h5>
    className
  </h5>
  <div
    style={
      Object {
        "border": undefined,
        "margin": 4,
        "minHeight": undefined,
        "padding": 4,
      }
    }
  >
    <button
      className="button"
      onClick={[Function]}
      variant="outlined"
    >
      GET VALUE
    </button>
  </div>
  <div
    style={
      Object {
        "border": "2px solid RosyBrown",
        "margin": 4,
        "minHeight": undefined,
        "padding": 4,
      }
    }
  >
    <pre
      className="value"
    >
      {
  bar: 8
}
    </pre>
  </div>
</div>
`;

exports[`../GetValue GetValue ({"id":"id","className":"className"}) should render 1`] = `
<div
  className="className"
  id="id"
  style={
    Object {
      "border": "2px solid black",
      "margin": 4,
      "minHeight": undefined,
      "padding": 4,
    }
  }
>
  <h5>
    id
  </h5>
  <div
    style={
      Object {
        "border": undefined,
        "margin": 4,
        "minHeight": undefined,
        "padding": 4,
      }
    }
  >
    <button
      className="button"
      onClick={[Function]}
      variant="outlined"
    >
      GET VALUE
    </button>
  </div>
  <div
    style={
      Object {
        "border": "2px solid RosyBrown",
        "margin": 4,
        "minHeight": undefined,
        "padding": 4,
      }
    }
  >
    <pre
      className="value"
    >
      {
  bar: 8
}
    </pre>
  </div>
</div>
`;

exports[`../GetValue GetValue ({"id":"id"}) should render 1`] = `
<div
  id="id"
  style={
    Object {
      "border": "2px solid black",
      "margin": 4,
      "minHeight": undefined,
      "padding": 4,
    }
  }
>
  <h5>
    id
  </h5>
  <div
    style={
      Object {
        "border": undefined,
        "margin": 4,
        "minHeight": undefined,
        "padding": 4,
      }
    }
  >
    <button
      className="button"
      onClick={[Function]}
      variant="outlined"
    >
      GET VALUE
    </button>
  </div>
  <div
    style={
      Object {
        "border": "2px solid RosyBrown",
        "margin": 4,
        "minHeight": undefined,
        "padding": 4,
      }
    }
  >
    <pre
      className="value"
    >
      {
  bar: 8
}
    </pre>
  </div>
</div>
`;

exports[`../GetValue GetValue ({}) should render 1`] = `
<div
  style={
    Object {
      "border": "2px solid black",
      "margin": 4,
      "minHeight": undefined,
      "padding": 4,
    }
  }
>
  <h5 />
  <div
    style={
      Object {
        "border": undefined,
        "margin": 4,
        "minHeight": undefined,
        "padding": 4,
      }
    }
  >
    <button
      className="button"
      onClick={[Function]}
      variant="outlined"
    >
      GET VALUE
    </button>
  </div>
  <div
    style={
      Object {
        "border": "2px solid RosyBrown",
        "margin": 4,
        "minHeight": undefined,
        "padding": 4,
      }
    }
  >
    <pre
      className="value"
    >
      {
  bar: 8
}
    </pre>
  </div>
</div>
`;

exports[`../GetValue should use the right versions of its dependencies 1`] = `
Object {
  "@material-ui/core": "^4.5.0",
  "prop-types": "^15.7.2",
  "react": "^16.10.1",
}
`;
